app/src/org/commcare/dalvik/activities/CommCareSetupActivity.java

```diff
@@ -244,7 +246,19 @@ private void uiStateScreenTransition() {
                 break;
 
             default:

```

Comment:
And better yet, rename UiState.advanced to UiState.choosing_url or something, because 'advanced' is a horrible name.